src/utils/AuthContext.jsx

```diff
@@ -0,0 +1,91 @@
+/* eslint-disable no-unused-vars */
+
+import { createContext, useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import {
+    createUserWithEmailAndPassword,
+    signInWithEmailAndPassword,
+    onAuthStateChanged,
+    signOut,
+    UserCredential,
+    User,
+    GoogleAuthProvider,
+    signInWithPopup
+} from 'firebase/auth';
+import { auth } from '.';
+
+export const AuthContext = createContext({
+    /**
+     * @type {User}
+     */
+    user: null,
+
+    /**
+     * @param {string} email
+     * @param {string} password
+     * @returns {Promise<UserCredential>}
+     */
+    register: (email, password) => Promise,
+
+    /**
+     * @param {string} email
+     * @param {string} password
+     * @returns {Promise<UserCredential>}
+     */
+    login: (email, password) => Promise,
+
+    /**
+     * @returns {Promise<UserCredential>}
+     */
+    loginWithGoogle: () => Promise,
+
+    /**
+     * @returns {Promise<void>}
+     */
+    logout: () => Promise
+});
+
+export const AuthContextProvider = ({ children }) => {
+    const [user, setUser] = useState(null);
+
+    useEffect(() => {
+        const unsubscribe = onAuthStateChanged(auth, (u) => {
+            setUser(u);
+        });
+        return unsubscribe;
+    }, []);
+
+    const register = (email, password) => {
```

Comment:
will we even have like custom register, i thought that we are going to stick to the just google auth